src/Elliptic/DiscontinuousGalerkin/SubdomainOperator/InitializeSubdomain.hpp

```diff
@@ -0,0 +1,521 @@
+          domain::Tags::Interface<domain::Tags::BoundaryDirectionsInterior<Dim>,
+                                  face_normal_magnitude_tag>,
+          ::Tags::Mortars<domain::Tags::Mesh<Dim - 1>, Dim>,
+          ::Tags::Mortars<::Tags::MortarSize<Dim - 1>, Dim>,
+          elliptic::dg::subdomain_operator::Tags::NeighborMortars<
+              domain::Tags::Mesh<Dim>, Dim>,
+          elliptic::dg::subdomain_operator::Tags::NeighborMortars<
+              face_normal_magnitude_tag, Dim>,
+          elliptic::dg::subdomain_operator::Tags::NeighborMortars<
+              domain::Tags::Mesh<Dim - 1>, Dim>,
+          elliptic::dg::subdomain_operator::Tags::NeighborMortars<
+              ::Tags::MortarSize<Dim - 1>, Dim>>>;
+
+  // Only slice those background fields to internal boundaries that are
+  // necessary for the DG operator, i.e. the background fields in the
+  // System::fluxes_computer::argument_tags
+  using fluxes_non_background_args =
+      tmpl::list_difference<typename System::fluxes_computer::argument_tags,
+                            typename System::background_fields>;
+  using background_fields_internal =
+      tmpl::list_difference<typename System::fluxes_computer::argument_tags,
+                            fluxes_non_background_args>;
+  using background_fields_overlap_internal = db::wrap_tags_in<
+      overlaps_tag,
+      tmpl::transform<
+          background_fields_internal,
+          make_interface_tag<tmpl::_1,
+                             tmpl::pin<domain::Tags::InternalDirections<Dim>>,
+                             tmpl::pin<tmpl::list<>>>>>;
+  // Slice all background fields to external boundaries for use in boundary
+  // conditions
+  using background_fields_external = typename System::background_fields;
+  using background_fields_overlap_external = db::wrap_tags_in<
+      overlaps_tag,
+      tmpl::transform<
+          background_fields_external,
+          make_interface_tag<
+              tmpl::_1,
+              tmpl::pin<domain::Tags::BoundaryDirectionsInterior<Dim>>,
+              tmpl::pin<tmpl::list<>>>>>;
+
+ public:
+  using initialization_tags =
+      tmpl::list<domain::Tags::InitialExtents<Dim>,
+                 domain::Tags::InitialRefinementLevels<Dim>>;
+  using const_global_cache_tags =
+      tmpl::list<LinearSolver::Schwarz::Tags::MaxOverlap<OptionsGroup>>;
+  using simple_tags = tmpl::append<
+      geometry_tags,
+      tmpl::conditional_t<has_background_fields,
+                          tmpl::list<overlaps_tag<::Tags::Variables<
+                              typename System::background_fields>>>,
+                          tmpl::list<>>,
+      background_fields_overlap_internal, background_fields_overlap_external>;
+  using compute_tags = tmpl::list<>;
+
+  template <
+      typename DataBox, typename... InboxTags, typename Metavariables,
+      typename ActionList, typename ParallelComponent,
+      Requires<tmpl::all<initialization_tags,
```

Comment:
yea this is an annoying thing that would be nice to sort out in the initialization actions code. The `if constexpr` would mean the entire function body must be indented by one level, or moved to a separate `_impl` function, which is not great either. It would be nice to get rid of this `Requires` entirely. I think this would be resolved by a non-looping initialization phase.

Reply:
I expect an indentation level is easier to understand for folks newer to C++ than the requires, though. I agree, it would be nice to get rid of the looping ability for certain phases in the long term

Reply:
ok, squashed in this change